sdk/examples/xo_python/sawtooth_xo/processor/xo_payload.py

```diff
@@ -0,0 +1,72 @@
+# Copyright 2018 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# -----------------------------------------------------------------------------
+
+from sawtooth_sdk.processor.exceptions import InvalidTransaction
+
+
+class XoPayload(object):
+
+    def __init__(self, payload):
+        try:
+            # The payload is csv utf-8 encoded string
+            name, action, space = payload.decode().split(",")
+        except ValueError:
+            raise InvalidTransaction("Invalid payload serialization")
+
+        _validate_transaction(name, action, space)
+
+        if action == 'take':
+            space = int(space)
+
+        self._name = name
+        self._action = action
+        self._space = space
+
+    @staticmethod
+    def from_bytes(payload):
+        return XoPayload(payload=payload)
+
+    @property
+    def name(self):
+        return self._name
+
+    @property
+    def action(self):
+        return self._action
+
+    @property
+    def space(self):
+        return self._space
+
+
+def _validate_transaction(name, action, space):
```

Comment:
This doesn't take a transaction, so validate_transaction is confusing. Would this be cleaner as a _validate() method on XoPayload or even inline in the XoPayload constructor?